Lib/fontTools/ttLib/tables/otBase.py

```diff
@@ -73,16 +76,34 @@ def compile(self, font):
 
 		# 		If a lookup subtable overflows an offset, we have to start all over.
 		overflowRecord = None
+		use_hb_repack = font.cfg[USE_HARFBUZZ_REPACKER]
+		if self.tableTag in ("GSUB", "GPOS"):
+			if not use_hb_repack:
+				log.debug(
+					"hb.repack disabled, compiling '%s' with pure-python serializer",
+					self.tableTag,
+				)
+			elif not have_uharfbuzz:
+				log.debug(
+					"uharfbuzz not found, compiling '%s' with pure-python serializer",
+					self.tableTag,
+				)
 
 		while True:
 			try:
 				writer = OTTableWriter(tableTag=self.tableTag)
 				self.table.compile(writer, font)
-				if have_uharfbuzz and self.tableTag in ("GSUB", "GPOS"):
+				if use_hb_repack and have_uharfbuzz and self.tableTag in ("GSUB", "GPOS"):
```

Comment:
Should it raise an error if `use_hb_repack` is True and `have_uharfbuzz` is False?

Reply:
No, I think it should not. USE_HARFBUZZ_REPACKER is either True or False, and it's True by default. What it does is it always tries to use uharfbuzz when importable, and if the latter errors out, falls back to pure-python serializer. Setting the option to False simply means ignore whether uharfbuzz is importable or not, just use the pure-python serializer all the time.
If we where to raise when USE_HARFBUZZ_REPACKER is True and have_uharfbuzz is False, then we'd need a tri-state boolean, defaulting to None, to distinguish the case when we try to use hb if available (default), or we request to use it and error out if not available (True).

Reply:
Or we would have to change the default to False, so it'd only work if one explicitly sets it to True

Reply:
alright, i'll make it a 3-state variable

Reply:
Done